core/typescript/src/lib/metadata.ts

```diff
@@ -185,9 +206,10 @@ export interface ConstructsInfo {
    */
   readonly relations: RelationsInfo;
   /**
-   * Information about custom (code blocks).
+   * Information about optional custom (code blocks).
+   * Missing this field implies that the package does NOT support custom code blocks.
    */
-  readonly custom: CustomInfo;
+  readonly custom?: CustomInfo;
```

Comment:
Custom code block as a feature should be optional as a way to limit capabilities offered by a package. A case in point is to allow only service/component customisation in k8s using cdk8s